### PR TITLE
Upgrade terraform-provider-azuredevops to v0.5.0

### LINKED ISSUE #151: Upgrade terraform-provider-azuredevops to v0.5.0
Release details: https://github.com/microsoft/terraform-provider-azuredevops/releases/tag/v0.5.0

### PR DESCRIPTION
**Pull request by aq17**
This PR was generated via `$ upgrade-provider pulumi/pulumi-azuredevops --upstream-provider-name=terraform-provider-azuredevops`.

---

Upgrading terraform-provider-azuredevops from 0.4.0  to 0.5.0.
Fixes #151
Upgrading pulumi-terraform-bridge from v3.42.1 to v3.46.1.

### CODE REVIEW

**Comment by aq17** [comment]
Changes look like they align with [upstream release notes](https://github.com/microsoft/terraform-provider-azuredevops/releases
)

File: provider/go.mod [hunk 1]
```diff
@@ -4,13 +4,13 @@ go 1.19
 
 replace (
 	github.com/hashicorp/go-getter v1.5.0 => github.com/hashicorp/go-getter v1.4.0
-	github.com/hashicorp/terraform-plugin-sdk/v2 => github.com/pulumi/terraform-plugin-sdk/v2 v2.0.0-20221122203342-430f685de305
+	github.com/hashicorp/terraform-plugin-sdk/v2 => github.com/pulumi/terraform-plugin-sdk/v2 v2.0.0-20230327102345-3fa930f86570
```
Comment (t0yv0) [inline provider/go.mod:7]: It's this one? https://github.com/pulumi/terraform-plugin-sdk/tree/upstream-v2.26.1 ? 

Curious.

**Review by t0yv0 — APPROVED** [review]
Looks clean!

File: provider/go.mod (hunk 1 above)
Comment (aq17) [reply provider/go.mod:7]: Do you mean the cause of the breaking changes?

Comment (t0yv0) [reply provider/go.mod:7]: No, I was curious why the hash of the github.com/pulumi/terraform-plugin-sdk fork is changing and is it in fact an upgrade; looks like it is, looks like Ian pushed a more recent branch there. So all good, just not something I noticed before.

**Pull request merged**